tests/threadcxx/enc/enc.cpp

```diff
@@ -89,6 +89,7 @@ static void _test_mutex1_cxx(size_t* count)
     std::stringstream ss;
 
     mutex1.lock();
+    OE_TEST(!mutex1.try_lock());
```

Comment:
According to standard, just like lock, try_lock also has undefined behavior:
https://en.cppreference.com/w/cpp/thread/mutex/try_lock

Reply:
I was based on https://linux.die.net/man/3/pthread_mutex_trylock, which mentions
> The pthread_mutex_trylock() function shall be equivalent to pthread_mutex_lock(), except that if the mutex object referenced by mutex is currently locked (by any thread, including the current thread), the call shall return immediately.

Also I just come across the code snippet in https://www.enseignement.polytechnique.fr/informatique/INF478/docs/Cpp/en/cpp/thread/mutex/unlock.html,
which also uses try_lock to determine if the mutex is locked or not.

Reply:
I don't think that is standards conforming behavior. A precondition for both lock and unlock operations of a mutex type is that the calling thread does not own the mutex:
http://www.eel.is/c++draft/thread.mutex.requirements

Reply:
I agree with it.
cc @achamayou @eddyashton 
Can you provide inputs on what's the expected behavior you want? It appears that it's inappropriate to reply on trylock() to test std::mutex.

Reply:
Removed the use of try_lock() in the tests.